kusto/ingest/ingest.go

```diff
@@ -119,52 +118,60 @@ func FlushImmediately() FileOption {
 // Not all options can be used in every method.
 type DataFormat = properties.DataFormat
 
+// [Yochai 15.06.2020]
+// John, I want to delete this and use properties.DataFormat insetad, but I get the feeling this will cause a breaking change.
+// WDYT?
+
 // note: any change here needs to be kept up to date with the properties version.
 // I'm not a fan of having two copies, but I don't think it is worth moving to its own package
 // to allow properties and ingest to both import without a cycle.
 const (
 	// DFUnknown indicates the EncodingType is not set.
-	DFUnknown DataFormat = 0
-	// CSV indicates the source is encoded in comma seperated values.
-	CSV DataFormat = 1
-	// JSON indicates the source is encoded in Javscript Object Notation.
-	JSON DataFormat = 2
+	DFUnknown DataFormat = properties.DFUnknown
 	// AVRO indicates the source is encoded in Apache Avro format.
-	AVRO DataFormat = 3
-	// Parquet indicates the source is encoded in Apache Parquet format.
-	Parquet DataFormat = 4
+	AVRO DataFormat = properties.AVRO
+	// APACHE AVRO indicates the source is encoded in Apache avro2json format.
+	APACHEAVRO DataFormat = properties.APACHEAVRO
+	// CSV indicates the source is encoded in comma seperated values.
+	CSV DataFormat = properties.CSV
+	// JSON indicates the source is encoded as one or more lines, each containing a record in Javscript Object Notation.
+	JSON DataFormat = properties.JSON
+	// JSON indicates the source is encoded in JSON-Array of individual records in Javscript Object Notation.
+	MULTIJSON DataFormat = properties.MULTIJSON
 	// ORC indicates the source is encoded in Apache Optimized Row Columnar format.
-	ORC DataFormat = 5
+	ORC DataFormat = properties.ORC
+	// Parquet indicates the source is encoded in Apache Parquet format.
+	Parquet DataFormat = properties.Parquet
 	// PSV is pipe "|" separated values.
-	PSV DataFormat = 6
+	PSV DataFormat = properties.PSV
 	// Raw is a text file that has only a single string value.
-	Raw DataFormat = 7
+	Raw DataFormat = properties.Raw
 	// SCSV is a file containing semicolon ";" separated values.
-	SCSV DataFormat = 8
+	SCSV DataFormat = properties.SCSV
 	// SOHSV is a file containing SOH-separated values(ASCII codepont 1).
-	SOHSV DataFormat = 9
-	// TSV is a file containing table seperated values ("\t").
-	TSV DataFormat = 10
+	SOHSV DataFormat = properties.SOHSV
+	// SINGLEJSON indicates the source containg a single Javascript Object Notation encoded record, newlines are treated as whitespace
+	SINGLEJSON DataFormat = properties.SINGLEJSON
+	// SSTREAM indicats the source is encoded as a Microsoft Cosmos Structured Streams format
+	SSTREAM DataFormat = properties.SSTREAM
```

Comment:
IIRC, SStream & WCLOGFILE are MS internal, do we want to include them in our SDKs?

Reply:
Hi @vladikbr, 
IIRC, you asked to add all types. 
Are there internal, redundant or soon to be deprecated formats which should not be included?

Reply:
As discussed, keeping all but Single Json